compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/AnnotationDesugar.java

```diff
@@ -628,6 +1955,44 @@ private BLangLambdaFunction defineAnnotations(BLangTypeDefinition typeDef, BLang
         return lambdaFunction;
     }
 
+
+    public BLangLambdaFunction defineAnnotationsForLocalRecords(BLangRecordTypeNode recordTypeNode,
```

Comment:
Why do we need a separate method for local anonymous records? Why can't we make the existing method work?

Reply:
?

Reply:
Refactor the code. please check